rfc-029-repositories-should-be-public-by-default.md

```diff
@@ -0,0 +1,81 @@
+# Source code repositories should be public by default
+
+## Summary
+
+We believe in working in the open and sharing our work with others. We think
+that nothing should be secret unless there’s a good reason for it to be. This
+is also a belief of central government technology practice, as described in the
+design principle “Make things open: it makes things better”.
+
+All GitHub repositories created by dxw which don’t directly relate to a client
+project SHOULD be public.
+
+When starting a new client project, we SHOULD encourage the client to allow us
+to make the repository public.
+
+## Problem
+
+- dxw creates a lot of source code repositories – for tools that we use
+  internally, and for client work.
+- When writing new code, we have a choice between making it available to the
+  public and keeping it private for our own use or that of the client.
+- Sometimes the code that we write might be useful to the rest of the world –
+  it solves a common problem, it provides inspiration for how to solve a
+  problem
+- We don’t usually work on projects that require secrecy.
+- We already make most of our repositories public – most notably the Playbook –
+    but until now we haven’t had an official policy, although we’ve [tried
+  to](https://github.com/dxw/playbook/issues/12) in the past. Most of our client
+  projects are also in public repositories.
+- We’ve [already talked
+  about](rfc-018-our-decision-making-process-is-public.md) why we make our RFC
+  decision making process open – the same reasons apply here
+
+Allowing the team to work in the open also has benefits for individual
+developers. They:
+
+- can show their work to friends and family – show the thing instead of talking
+  about the thing
+- know that they’re investing their energy in something maximally useful and
+  long-lasting
+- can build up a public portfolio of work
+
+## Proposal
+
+All new GitHub respositories in the dxw organisation SHOULD be public, unless
+they directly relate to a client project.
+
+- TODO – I can’t think of many reasons why such a repo shouldn’t be public –
+  perhaps it contains secrets e.g. keys? Bring this up at tech team forum
```

Comment:
Repos should never contain secrets. Making them public is a good way to avoid the temptation to do this :)